pkg/csi/service/wcp/controller.go

```diff
@@ -1697,6 +1739,26 @@ func (c *controller) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshot
 	return resp, err
 }
 
+func patchCNSVolumeInfoWithRetry(ctx context.Context, volumeID string, patchBytes []byte) error {
```

Comment:
Let's update the existing method to achieve this, maybe make retry as a bool parameter.